### PR TITLE
[DeadCode] Add RemoveDeadInstanceOfRector

### LINKED ISSUE #5458: [DeadCode] Remove check exact value instanceof on type hinted ClassMethod
# Feature Request

To remove unneded check same exact value instanceof like the following code:

https://github.com/rectorphp/rector/blob/e9cdf37d160ea7dab2a7662c95508ec2d1db2988/rules/renaming/src/Rector/MethodCall/RenameMethodRector.php#L125-L131

The `$classMethod` is a `ClassMethod`, no need to check if `$classMethod` is a `ClassMethod`

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #5458

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
See `\Rector\Core\Rector\AbstractTemporaryRector::unwrapStmts()` for unwrapping many stmts 

```diff
-if (...) {
   $stmt;
   $stmt2;
-}
```

**Comment by samsonasik** [comment]
implemented with `unwrapStmts()`

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/dead-code/tests/Rector/If_/RemoveDeadInstanceOfRector/Fixture/do_not_remove_parent_child_flip.php.inc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        if (! $var instanceof If_) {
+            return false;
+        }
+        return true;
+    }
+
+    public function run(If_ $var)
+    {
+        return true;
+    }
```
Comment (TomasVotruba) [inline rules/dead-code/tests/Rector/If_/RemoveDeadInstanceOfRector/Fixture/do_not_remove_parent_child_flip.php.inc:49]: I love how you made it advanced with sub types 👏

**Comment by TomasVotruba** [comment]
Thanks

**Pull request merged**